drivers/bluetooth/hci/h4.c

```diff
@@ -523,6 +524,14 @@ static int h4_open(const struct device *dev, bt_hci_recv_t recv)
 
 	uart_irq_callback_user_data_set(cfg->uart, bt_uart_isr, (void *)dev);
 
+#if DT_NODE_HAS_PROP(DT_DRV_INST(0), reset_gpios)
+	const struct gpio_dt_spec rst_gpio = GPIO_DT_SPEC_INST_GET(0, reset_gpios);
```

Comment:
Even though the Zephyr host stack currently only supports a single instance, the HCI driver API and most drivers where it's possible, including this `h4.c`, support multi instance. There's also some interest/requests to add multi-instance support for the host: #93641 (although I'm a bit skeptical if that'll ever materialize).

Due to the above, I think it'd be better to do this properly here in `h4.c` and add the info to `dev->config` (i.e. `struct h4_config`). To avoid overhead when no instance uses this feature, you could use a macro like `DT_ANY_COMPAT_HAS_PROP_STATUS_OKAY()`.